docs/02-developers/05-smart-contracts/04-verify-smart-contracts/Verify-Smart-Contracts-using-Blockscout/index.md

```diff
@@ -0,0 +1,181 @@
+---
+id: verify-foundry-contract
+sidebar_label: Verify Smart Contracts using Blockscout (Foundry)
+sidebar_position: 400
+title: Verify a Smart Contract using Blockscout (Foundry)
+description: "Configuring Foundry for smart contract verification on Rootstock"
+tags: [guides, developers, smart contracts, rsk, rootstock, foundry, dApps, ethers]
+---
+
+
+
+Smart contracts are the backbone of decentralized applications (dApps). They automate agreements and processes, but their code can be complex and prone to errors. Verifying your smart contracts is crucial to ensure they function as intended.
+
+This tutorial will guide you through verifying your contracts using Foundry on the Rootstock Blockscout Explorer. Foundry is a powerful toolkit spektrum for Ethereum development, and its `forge` tool simplifies the verification of Solidity smart contracts deployed on the Rootstock network. By verifying the contracts, you allow Blockscout, an open-source block explorer, to link your contract's source code with its deployed bytecode on the blockchain, enabling trustless interaction with the code.
+
+In this tutorial, we'll do the following steps:
+
+- Set up your Foundry configuration environment in your project
+- Use Foundry's `forge verify-contract` to verify a contract address
+
+## Prerequisites
+
+To follow this tutorial, you should have knowledge of the following:
+
+- Foundry
+- Basic knowledge of smart contracts
+
+:::note[Foundry Starter Project]
+A Foundry starter project can be set up with preset configurations for the Rootstock network. Initialize a new Foundry project using `forge init` and configure it as shown below. Ensure you set up the `.env` variables to match the `foundry.toml` configuration.
+:::
+
+To add a step-by-step guide for obtaining a Rootstock API key from Blockscout, you can insert the following section into the tutorial, ideally after the **Prerequisites** section and before the **Configuration** section, as it logically fits where users need to prepare their API key for the `foundry.toml` configuration. Below is the proposed addition, formatted to match the tutorial's style and tone:
+
+---
+
+## Obtaining a Rootstock API Key from Blockscout
+
+To verify smart contracts on the Rootstock Blockscout Explorer, you need an API key. Follow these steps to obtain one:
+
+1. **Visit the Blockscout Explorer**:
+   - For Rootstock Mainnet, go to [https://rootstock.blockscout.com/](https://rootstock.blockscout.com/).
```

Comment:
- For [Rootstock Mainnet](https://rootstock.blockscout.com/)
- For [Rootstock Testnet](https://rootstock-testnet.blockscout.com/)